Bottom: 2b9e65eb6d97cafe740e5938de18ddf62a16f5bf
Top:    e44efa8125faed7e5b04bf16cbe9bea6404801dc
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-10 16:31:11 +0000

Revert "Move gdk_window_process_all_updates to safest spot"

This reverts commit f876b22592f566deccb35e9aface6df4c2dfffa9.

Because it might be a tiny bit inefficient for zoom where it was.  This
way nothing hits this except the case where we're about to do a pointer
warp


---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 1e210ab..a4ece3a 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -39,6 +39,7 @@ pan_common (GHidPort *port)
   ghidgui->adjustment_changed_holdoff = FALSE;
 
   ghid_port_ranges_changed();
+  gdk_window_process_all_updates ();
 }
 
 static void
@@ -335,10 +336,6 @@ ghid_set_crosshair (int x, int y, int action)
       ghid_pan_view_abs (gport->crosshair_x - gport->view.width / 2,
                          gport->crosshair_y - gport->view.height / 2,
                          0, 0);
-  
-      // We do this to make sure gdk has an up-to-date idea of the widget
-      // coordinates so gdk_display_warp_pointer will go to the right spot.
-      gdk_window_process_all_updates ();
 
       // Warp pointer to crosshair
       ghid_pcb_to_event_coords (gport->crosshair_x, gport->crosshair_y,
